gwt-ol3-client/src/main/java/ol/MapOptions.java

```diff
@@ -32,49 +35,110 @@
+     * requiring that the target element has a <code>tabindex</code> attribute.
+     * 
+     * @param target The id of the element to listen keyboard events on.
      */
     @JsProperty
-    public native void setControls(Collection<Control> controls);
+    public native void setKeyboardEventTarget(String target);
 
     /**
-     * Set the map's view. No layer sources will be fetched unless this is
-     * specified at construction time or through ol.Map#setView.
-     *
-     * @param view {@link View}
+     * The element to listen to keyboard events on. This determines when the {@link ol.interaction.KeyboardPan} and {@link ol.interaction.KeyboardZoom} interactions trigger.
+     * For example, if this option is set to <code>document</code> the keyboard interactions will always trigger.
+     * If this option is not specified, the element the library listens to keyboard events on is the map target
+     * (i.e. the user-provided div for the map). If this is not <code>document</code> the target element needs to be focused for key events to be emitted,
+     * requiring that the target element has a <code>tabindex</code> attribute.
+     * 
+     * @param target The element to listen keyboard events on.
      */
     @JsProperty
-    public native void setView(View view);
+    public native void setKeyboardEventTarget(Element target);
 
+    /**
+     * The element to listen to keyboard events on. This determines when the {@link ol.interaction.KeyboardPan} and {@link ol.interaction.KeyboardZoom} interactions trigger.
+     * For example, if this option is set to <code>document</code> the keyboard interactions will always trigger.
+     * If this option is not specified, the element the library listens to keyboard events on is the map target
+     * (i.e. the user-provided div for the map). If this is not <code>document</code> the target element needs to be focused for key events to be emitted,
+     * requiring that the target element has a <code>tabindex</code> attribute.
+     * 
+     * @param target The document to listen keyboard events on.
+     */
     @JsProperty
-    public native void setInteractions(Interaction[] interactions);
+    public native void setKeyboardEventTarget(Document target);
 
-	/**
-	 * Interactions that are initially added to the map. If not specified,
-	 * ol.interaction.defaults() is used.
-	 *
-	 * @param interactions
-	 */
-	@JsProperty
-	public native void setInteractions(Collection<Interaction> interactions);
+    /**
+     * Layers. If this is not defined, a map with no layers will be rendered. Note that layers are rendered in the order supplied,
+     * so if you want, for example, a vector layer to appear on top of a tile layer, it must come after the tile layer.
+     * 
+     * @param layers Layers.
+     */
+    @JsProperty
+    public native void setLayers(Base[] layers);
```

Comment:
This is not my code, I just moved it around so git got confused.

If either Collections or arrays are preferable, I can remove other setters.